### DISCUSSION TITLE
Using FunctionChoiceBehavior auto function calling filters to exclude functions

### DISCUSSION THREAD

**Discussion by ColtonBehannon** [microsoft/semantic-kernel#7600] [Q&A] [answered]
I have an application that consists of multiple agents (I am not using the agents abstraction) that differ from each other only in the plugins they (should) have available to them. One is a general agent that can access all internet sites and functions via the WebScraper plugin I have created. The other agent should only have access to two of the functions within the WebScraper plugin which restrict it to news sites.

I swap between the two agents by changing the `FunctionChoiceBehavior.Auto(filters={})` from:
`filters={"excluded_plugins": ["ChatBot"]}`
to 
`filters={"excluded_plugins": ["ChatBot"], "excluded_functions": ["GeneralWebSearch", "GeneralWebSearch_ReliableSources"]}`

I do this in an attempt to remove the two functions that I do not want it to have access to. However, my agent is still able to make calls to both of the functions listed within the excluded_functions. Am I misunderstanding how to use this?

**Comment by moonbox3** [comment, marked as answer]
Hi @ColtonBehannon, thanks for the question. If you want to exclude a function via the filters dictionary, you'll need to make sure you're supplying the correct fully qualified name. We use a hyphen ("-") to separate plugin names and function names. Therefore if you're hoping to exclude the plugin `GeneralWebSearch` with the function name `ReliableSources`, it should be written as `GeneralWebSearch-ReliableSources`. 

This is the code that is being called to filter out plugins or functions:

https://github.com/microsoft/semantic-kernel/blob/a53b96a1ae066160872b3cad65387442995a788d/python/semantic_kernel/functions/kernel_function_extension.py#L368C5-L410C22

**Reply by ColtonBehannon** [reply]
I knew there had to be something simple I was overlooking. It is working great now, thank you!
